docs/conceptual/cryptography/keys.md

```diff
@@ -0,0 +1,145 @@
+default. In this system, there are two core roles:
+
+1. Validators: Users that produce blocks.
+2. Nominators: Users that support a validator.
+
+Validators need to collect funds which back their intention to produce blocks.
+Validators can provide their own funds, and also collect funds from nominators
+who trust the validator to produce new blocks correctly. As a reward for doing
+things correctly, both validators and their respective nominators get paid.
+These funds can be used for slashing if the validator misbehaves.
+
+> **Note:** Learn more about validators and nominators in the context of the
+> Substrate's NPoS [Staking module](/rustdocs/master/srml_staking/index.html).
+
+Both validators and nominators may be working with a significant value of funds,
+so Substrate has introduced abstractions which allow safer behaviors for end
+users participating in the network.
+
+These abstractions are:
+
+* [Account Keys](#account-keys): Accounts that should hold funds.
+    * [Stash Keys](#stash-keys): Cold wallets controlled by controller keys.
+    * [Controller Keys](#controller-keys): Semi-online keys that will be in the
+      direct control of a user and used to submit extrinsics.
+* [Session Keys](#session-keys): Hot keys that be must kept online by a
+  validator to perform network operations and should not hold funds.
+
+## Account Keys
+
+Account keys represent the normal kind of accounts you might expect from other
+blockchain systems. These accounts hold funds, and in the context of Substrate's
+Balances module, must have at least an existential deposit amount of those
+funds. Accounts keys are defined generically and its type is ultimately made
+concrete in the runtime. They interact with the blockchain through transactions.
+
+In the context of Substrate's Staking module, there are two abstractions on top
+of account keys:
+
+1. Stash Keys
+2. Controller Keys
+
+These keys are distinguished by their intended use, not by any underlying
+cryptographic difference. When creating new controller or stash keys, all
+cryptography supported by account keys are an available option.
+
+### Stash Keys
+
+Stash keys are intended to behave like a cold wallet, existing on a piece of
+paper in a safe or protected by layers of hardware security. It should rarely,
+if ever, be exposed to the internet or used to submit extrinsics. They are meant
+to hold a large amount of funds, and should be thought of as a saving's account
+at a bank, which ideally is only ever touched in urgent conditions.
+
+Since the stash key is kept offline, it must be set to have its funds bonded to
+a controller account. For non-spending actions, the controller has the funds of
+the stash behind it. For example, in nominating, staking, or voting, the
+controller can indicate its preference with the weight of the stash.
+
+### Controller Keys
+
+The controller key is a semi-online key that will be in the direct control of a
```

Comment:
The Controller keys are the public/private key pair that defines a Controller account. In the context of Substrate's NPoS model, the Controller key will signal one's intent to validate or nominate.

The Controller key is used to set preferences like rewards destination and, in the case of validators, to set their Session keys. The Controller account only needs to pay transaction fees, so needs a minimal amount of funds.

The Controller key can never be used to spend funds from its Stash account. However, actions taken by the Controller can result in slashing, so it should still be well secured.